### PR TITLE
Add some components to the new testing framework (H)

### PR DESCRIPTION
**Pull request by kbx81**
# What does this implement/fix?

Add components beginning with the letter "H" to the new testing framework.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: tests/components/homeassistant/test.esp32.yaml
Comment (jesserockz) [inline tests/components/homeassistant/test.esp32.yaml:1]: Future PR note:
These tests should also include `homeassistant.event` and `homeassistant.service` actions and user defined api services

File: tests/components/http_request/test.esp32.yaml
Comment (jesserockz) [inline tests/components/http_request/test.esp32.yaml:1]: Future PR note:
Add actions to these tests

**Review by jesserockz — APPROVED** [review]

File: tests/components/homeassistant/test.esp32.yaml
Comment (kbx81) [reply tests/components/homeassistant/test.esp32.yaml:1]: https://github.com/esphome/esphome/pull/6528

File: tests/components/http_request/test.esp32.yaml
Comment (kbx81) [reply tests/components/http_request/test.esp32.yaml:1]: https://github.com/esphome/esphome/pull/6529

**Pull request merged**